File: LeetCode_Solutions/LeetCode_Solutions/BestTimeToBuyAndSellStock_121.m
//
//  BestTimeToBuyAndSellStock_121.m
//  LeetCode_Solutions
//
//  Created by 来自遥远星系的核心巡洋舰 on 16/11/9.
//  Copyright © 2016年 KnightJoker. All rights reserved.
//

#import "BestTimeToBuyAndSellStock_121.h"

@implementation BestTimeToBuyAndSellStock_121

- (NSInteger)maxProfit:(NSArray *)prices {


    NSInteger temp = [prices[0] integerValue];
    NSInteger dif = 0;
    
    for (NSInteger i = 1; i < prices.count; i++) {
        if ([prices[i] integerValue] < temp) {
            temp = [prices[i] integerValue];
        } else if ([prices[i] integerValue] - temp > dif) {
            dif = [prices[i] integerValue] - temp;
        }
    }
    
    return dif;
}

@end

File: LeetCode_Swift/LeetCode_Swift/BestTimeToBuyAndSellStock_121.swift
//
//  BestTimeToBuyAndSellStock_121.swift
//  LeetCode_Swift
//
//  Created by Huni on 21/08/2017.
//  Copyright © 2017 KnighhtJoker. All rights reserved.
//

import Foundation


public class BestTimeToBuyAndSellStock_121 {

    func maxProfit(_ prices: [Int]) -> Int {
        
        if (prices.count < 1) {
            return 0;
        }
        
        var temp = prices[0]
        var dif = 0
        var index = 1
        
        while index < prices.count {
            if prices[index] < temp {
                temp = prices[index]
            } else if prices[index] - temp > dif {
                dif = prices[index] - temp
            }
            index += 1
        }
        
        return dif
    }
}
